### ISSUE TITLE
Update indent rule behavior to match Xcode 12

### ISSUE THREAD

**Issue by nicklockwood** [nicklockwood/SwiftFormat#744]
Xcode 12 adds new alignment rules for wrapped expressions, e.g.

```swift
if let foo = bar,
    let baz = quux {

}
```

Now aligns the `let`s vertically:

```swift
if let foo = bar,
   let baz = quux {

}
```

Which means that the second line is actually indented exactly 3 spaces and not a multiple of the specified indent level.

SwiftFormat should probably adopt this, especially when using `--xcodeindentation true`. There's a question of whether this should have some way to retain the old behavior though for teams that haven't upgraded to Xcode 12. 

@calda, any thoughts?

**Comment by calda** [comment]
I would love to see this be an option even for teams that aren't using `--xcodeindentation true`!  

[We use](https://github.com/airbnb/swift/blob/master/resources/airbnb.swiftformat) `--indent 2` and `--xcodeindentation disabled`, but if there was a `--multilineConditional alignKeywords` option that works like this then I would totally lobby for us to adopt it.

**Comment by georgescumihai** [comment]
This is the same for `guard`, it will align it vertically

#### Xcode 12 indent
```
guard let foo = bar,
      let baz = quux else {
    return
}
```

#### swiftformat indent
```
guard let foo = bar,
    let baz = quux else {
        return
}
```

Also with swiftformat the `return` in case of `guard`  is alligned at 8 spaces comparing to 4 with Xcode 12.
For `if` statement, both Xcode 12 and swiftformat allign the return at 4 spaces.

**Comment by nicklockwood** [comment]
Implemented in 0.47.0

**Comment by georgescumihai** [comment]
Nice job, just tested it works, in most of the cases.
There are still situations where Xcode indentation is not following their own rules

#### Xcode
```
guard let value = foo
        .doSomething()
        .doSomethingElse() else {
    return
}
```

#### Swiftformat
```
guard let value = foo
    .doSomething()
    .doSomethingElse() else {
    return
}
```

Also `} else if let` does vertical aligning
#### Xcode
```
if foo {

} else if let bar = guas,
          let xuas = assas {

}

#### Swiftformat
```
if foo {

} else if let bar = guas,
    let xuas = assas {

}
```

**Comment by georgescumihai** [comment]
@nicklockwood maybe the thing that is missing is the vertical alignment for `else if let`

**Comment by nicklockwood** [comment]
@georgescumihai thanks, I'll fix  these cases in the next update

**Comment by georgescumihai** [comment]
For my tests to check the differences I use [SourceKitten](https://github.com/jpsim/SourceKitten) to interact with [SourceKit](https://github.com/apple/swift/tree/main/tools/SourceKit).
```
find . -type f -name "*.swift" | while read line; do
  sourcekitten format --file "$line";
done
```

**Comment by nicklockwood** [comment]
Fixed in 0.47.1
